lib/options.go

```diff
@@ -500,11 +500,38 @@ func (o Options) ForEachSpecified(structTag string, callback func(key string, va
 	for i := 0; i < structType.NumField(); i++ {
 		fieldType := structType.Field(i)
 		fieldVal := structVal.Field(i)
+		value := fieldVal.Interface()
 
 		shouldCall := false
 		switch fieldType.Type.Kind() {
 		case reflect.Struct:
+			// Unpack any guregu/null values
 			shouldCall = fieldVal.FieldByName("Valid").Bool()
+			valOrZero := fieldVal.MethodByName("ValueOrZero")
+			if shouldCall {
+				if valOrZero.IsValid() {
+					val := valOrZero.Call([]reflect.Value{})[0]
```

Comment:
Is `[0]` safe here or should we protect against `index out of range` errors?

Reply:
It should be fine, after all, we know that all `ValueOrZero()` calls return a result. If that changes in the future, it's better to panic instead of silencing the error.